tests/unit/test_path_template.py

```diff
@@ -387,3 +387,36 @@ def helper_test_transcode(http_options_list, expected_result_list):
         expected_result["body"] = expected_result_list[2]
 
     return (http_options, expected_result)
+
+
+def test_path_template_to_regex():
+    path_pattern_to_regex = {
+        "*": "^[^/]+$",
+        "**": "^.*$",
+        "foo": "^foo$",
+        "hello/world": "^hello/world$",
+        "hello/*": "^hello/[^/]+$",
+        "*/world": "^[^/]+/world$",
+        "hello/*/world": "^hello/[^/]+/world$",
+        "hello/*/**": "^hello/[^/]+(?:/.*)?$",
+        "hello/**": "^hello(?:/.*)?$",
+        "{foo}": "^(?P<foo>[^/]+)$",
+        "{foo=*}": "^(?P<foo>[^/]+)$",
+        "{foo=**}": "^(?P<foo>.*)$",
+        "{database=projects/*/databases/*}/documents/*/**": "^(?P<database>projects/[^/]+/databases/[^/]+)/documents/[^/]+(?:/.*)?$",
+        "documents/*/{database=projects/*/databases/*}/pages/*/**": "^documents/[^/]+/(?P<database>projects/[^/]+/databases/[^/]+)/pages/[^/]+(?:/.*)?$",
+        "": "^$",
+    }
+    for k, v in path_pattern_to_regex.items():
+        got = path_template.to_regex(k).pattern
+        assert got == v
```

Comment:
This would be clearer as a parameterized test, e.g.:
```suggestion
@pytest.mark.parametrize("template, regex_pattern", [
    ("", "^$"),
    ("*", "^[^/]+$"),
    ("**", "^.*$"),
    ("foo", "^foo$"),
    ("hello/world", "^hello/world$"),
    ("hello/*", "^hello/[^/]+$"),
    ("*/world", "^[^/]+/world$"),
    ("hello/*/world", "^hello/[^/]+/world$"),
    ("hello/*/**", "^hello/[^/]+(?:/.*)?$"),
    ("hello/**", "^hello(?:/.*)?$"),
    ("{foo}", "^(?P<foo>[^/]+)$"),
    ("{foo=*}", "^(?P<foo>[^/]+)$"),
    )"{foo=**}", "^(?P<foo>.*)$"),
    ("{database=projects/*/databases/*}/documents/*/**",
     "^(?P<database>projects/[^/]+/databases/[^/]+)/documents/[^/]+(?:/.*)?$"),
    ("documents/*/{database=projects/*/databases/*}/pages/*/**",
     "^documents/[^/]+/(?P<database>projects/[^/]+/databases/[^/]+)/pages/[^/]+(?:/.*)?$"),
])
def test_path_template_to_regex(template, regex_pattern):
    path_pattern_to_regex = {
        "*": "^[^/]+$",
        "**": "^.*$",
        "foo": "^foo$",
        "hello/world": "^hello/world$",
        "hello/*": "^hello/[^/]+$",
        "*/world": "^[^/]+/world$",
        "hello/*/world": "^hello/[^/]+/world$",
        "hello/*/**": "^hello/[^/]+(?:/.*)?$",
        "hello/**": "^hello(?:/.*)?$",
        "{foo}": "^(?P<foo>[^/]+)$",
        "{foo=*}": "^(?P<foo>[^/]+)$",
        "{foo=**}": "^(?P<foo>.*)$",
        "{database=projects/*/databases/*}/documents/*/**": "^(?P<database>projects/[^/]+/databases/[^/]+)/documents/[^/]+(?:/.*)?$",
        "documents/*/{database=projects/*/databases/*}/pages/*/**": "^documents/[^/]+/(?P<database>projects/[^/]+/databases/[^/]+)/pages/[^/]+(?:/.*)?$",
        "": "^$",
    }
    assert path_template.to_regex(template).pattern == regex_pattern

```